luxonis_train/attached_modules/losses/softmax_focal_loss.py

```diff
@@ -34,24 +34,43 @@ def __init__(
             C{0.25}.
         @type gamma: float
         @param gamma: Focusing parameter. Defaults to C{2.0}.
+        @type smooth: float
+        @param smooth: Label smoothing factor. Defaults to C{0.0}.
         @type reduction: Literal["none", "mean", "sum"]
         @param reduction: Reduction type. Defaults to C{"mean"}.
         """
         super().__init__(**kwargs)
 
-        self.alpha = alpha
+        self.alpha = alpha if alpha is not None else 1.0
         self.gamma = gamma
+        self.smooth = smooth
         self.reduction = reduction
-        self.ce_criterion = CrossEntropyLoss(reduction="none", **kwargs)
 
-    def forward(self, predictions: Tensor, target: Tensor) -> Tensor:
-        ce_loss = self.ce_criterion.forward(predictions, target)
-        pt = torch.exp(-ce_loss)
-        loss = ce_loss * ((1 - pt) ** self.gamma) * self.alpha
+        if self.smooth is not None:
+            if self.smooth < 0 or self.smooth > 1.0:
+                raise ValueError("smooth value should be in [0,1]")
+
+    def forward(
```

Comment:
Let's make sure that the output of this loss with smoothing turned off is same as with the previous implementation. And also let's make sure that it still works with e.g. segmentation model correctly.

We should keep naming of the parameters consistent across losses: `predictions`, `target`